test/others/loop.c

```diff
@@ -0,0 +1,40 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+	pid_t pid;
+	pid_t sid;
+
+	pid = fork();
+	if (pid < 0) {
+		perror("fork failed!");
+		return -1;
+	}
+
+	if (pid == 0) {
+		sid = setsid();
```

Comment:
but the parent process doesn't wait when a child will do setsid() and other actions. Did I miss something?

Reply:
I've updated the commit to wait for the child process to complete the setup.